lib/guid.ts

```diff
@@ -1,49 +1,72 @@
 export class Guid {
+    /** Empty GUID string (hyphenated). */
     public static EMPTY = "00000000-0000-0000-0000-000000000000";
 
     public static isValid(guid: any) {
         const value: string = guid.toString();
-        return guid && (guid instanceof Guid || Guid.validator.test(value));
+        return guid && (guid instanceof Guid || this.validator.test(value));
     }
 
     public static create(guid?: string): Guid {
-        if (!guid) { guid = [Guid.gen(2), Guid.gen(1), Guid.gen(1), Guid.gen(1), Guid.gen(3)].join("-"); }
+        if (!guid) guid = this.random();
         return new Guid(guid);
     }
 
+    /** Generates an empty, hyphenated GUID object. */
     public static createEmpty(): Guid {
-        return new Guid();
+        return new Guid(this.EMPTY);
     }
 
-    private static validator = new RegExp("^[a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12}$", "i");
+    private static validator: RegExp = new RegExp("^[a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12}$", "i");
 
-    private static gen(count: number) {
+    private static gen(count: number): string {
         let out: string = "";
         for (let i: number = 0; i < count; i++) {
             // tslint:disable-next-line:no-bitwise
-            out += (((1 + Math.random()) * 0x10000) | 0).toString(16).substring(1);
+            out += (((1 + Math.random()) * 0x10000) | 0).toString(16).substring(1).toLowerCase();
         }
         return out;
     }
 
+    /** Generates a random, hyphenated GUID string. */
+    public static random(): string {
+        return [Guid.gen(2), Guid.gen(1), Guid.gen(1), Guid.gen(1), Guid.gen(3)].join("-");
+    }
+
+    /** Container for the GUID itself (in the hyphenated format). */
     private value: string = Guid.EMPTY;
 
     constructor(guid?: string) {
-        if (guid && Guid.isValid(guid)) {
+        if (guid && Guid.isValid(guid)) { //hyphenated, no conversion necessary
             this.value = guid.toLowerCase();
-        }
+        } else if (guid && !guid.includes("-") && guid.length == 32) { //non-hyphenated
+            let tempGuid: string = "";
+            let pos: number = 0;
+            for (const i of [8, 4, 4, 4, 12]) {
```

Comment:
would be good if you would add an empty row for a better readability
```suggestion

            for (const i of [8, 4, 4, 4, 12]) {
```